modules/vault/main.tf

```diff
@@ -21,20 +21,17 @@ resource "azurerm_key_vault" "default" {
     default_action = "Allow"
   }
 
-  tags = {
-    "customer-ns" = var.namespace,
-    "env"         = "managed-install"
-  }
+  tags = var.tags
 }
 
 resource "azurerm_key_vault_access_policy" "parent" {
   key_vault_id = azurerm_key_vault.default.id
   tenant_id    = data.azurerm_client_config.current.tenant_id
   object_id    = data.azurerm_client_config.current.object_id
 
-  key_permissions     = ["Get", "Backup", "Delete", "List", "Purge", "Recover", "Restore", "Rotate", "GetRotationPolicy"]
-  secret_permissions  = ["Get", "Backup", "Delete", "List", "Purge", "Recover", "Restore", "Set"]
-  storage_permissions = ["Get", "Backup", "Delete", "List", "Purge", "Recover", "Restore"]
+  key_permissions     = ["Backup", "Create", "Decrypt", "Delete", "Encrypt", "Get", "GetRotationPolicy", "List", "Purge", "Recover", "Restore", "Rotate"]
+  secret_permissions  = ["Backup", "Delete", "Get", "List", "Purge", "Recover", "Restore", "Set"]
+  storage_permissions = ["Backup", "Delete", "Get", "List", "Purge", "Recover", "Restore"]
```

Comment:
could you please share the reason behind these additional permissions?

Reply:
These are required to permit access to the key in the key vault. I've already tagged @wandb-kc on this for sec review.